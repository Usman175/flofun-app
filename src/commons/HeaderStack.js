import React from 'react'
import { TouchableOpacity } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import styles from './styles/Header'

import Colors from '../../constants/colors'

const HeaderStack = (goBack, { hideBack } = {}) => {
  const style = { backgroundColor: Colors.purpleDarkColor }

  const titleStyle = {
    fontSize: 20,
    fontFamily: 'montserratBold',
    color: Colors.whiteColor,
    textAlign: 'center',
  }

  const left = hideBack ? null : (
    <TouchableOpacity style={styles.iconBack} onPress={() => goBack()}>
      <Ionicons
        name="ios-arrow-back"
        size={30}
        color="#fff"
      />
    </TouchableOpacity>
  )

  return { style, titleStyle, left }
}

export default HeaderStack
