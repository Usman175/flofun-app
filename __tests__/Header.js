import 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import Colors from '../constants/colors'
import HeaderStack from '../src/commons/HeaderStack'

beforeAll(() => {
  EStyleSheet.build(Colors)
})

describe('Header', () => {
  test('renders correctly', () => {
    expect(HeaderStack()).toMatchSnapshot()
  })
})
