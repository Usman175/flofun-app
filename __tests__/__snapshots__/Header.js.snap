// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Object {
  "headerLeft": <View
    style={
        Object {
            "flexDirection": "row",
          }
    }
>
    <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
            Object {
                "marginLeft": 11.25,
                "marginRight": 3.75,
              }
        }
    >
        <Icon
            allowFontScaling={false}
            color="#fff"
            name="ios-arrow-back"
            size={30}
        />
    </TouchableOpacity>
    <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
            Object {
                "marginLeft": 11.25,
              }
        }
    >
        <Icon
            allowFontScaling={false}
            color="#fff"
            name="menu"
            size={30}
        />
    </TouchableOpacity>
</View>,
  "headerStyle": Object {
    "backgroundColor": "#43405D",
  },
  "headerTitleStyle": Object {
    "color": "#fff",
    "fontFamily": "montserratBold",
    "fontSize": 20,
    "textAlign": "center",
  },
}
`;
